### PR TITLE
Allow for generating alternatively named SPM binaries with CMake

### PR DESCRIPTION
**Pull request by rjai**
This makes it less risky to package multiple SPM versions together.

### CODE REVIEW

**Review by snukky — COMMENTED** [review]
No need to have this alternative name configurable via command line for more flexibility? Otherwise, looks good to me. If you require a review from Marcin, please add him as a reviewer here. Even if that does speed things up, it will be clearer for others.

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,7 +15,13 @@
 cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 file(STRINGS "VERSION" SPM_VERSION)
 message(STATUS "VERSION: ${SPM_VERSION}")
+option(SPM_ALT_PROJNAME "Call sentencepiece sentencepiecelatest to avoid conflicts" OFF)
+
+if (SPM_ALT_PROJNAME)
+project(sentencepiecelatest VERSION ${SPM_VERSION} LANGUAGES C CXX)
```
[the remark this answers is not in the document]
Comment (rjai) [reply CMakeLists.txt]: Fixed

File: CMakeLists.txt [hunk 2]
```diff
@@ -15,7 +15,13 @@
 cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 file(STRINGS "VERSION" SPM_VERSION)
 message(STATUS "VERSION: ${SPM_VERSION}")
+option(SPM_ALT_PROJNAME "Call sentencepiece sentencepiecelatest to avoid conflicts" OFF)
```
[the remark this answers is not in the document]
Comment (rjai) [reply CMakeLists.txt]: Changed the message, and what the config parameter does appropriately.

**Comment by rjai** [comment]
Updated to not hard-code a particular alternative name and keep that configurable.

**Review by snukky — APPROVED** [review]

**Pull request merged**